### ISSUE TITLE
Creating a UI using streamlit for laptop price prediction model.

### ISSUE THREAD

**Issue by Gaurav-576** [abhisheks008/DL-Simplified#828]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : UI using Streamlit for Laptop Price Prediction</br>
:red_circle: **Aim** : To provide an interactive user experience and access the facilities of our ML model based on real life data for accurate prediction of laptop prices.</br>
:red_circle: **Dataset** : It does not contains any dataset as it is just a streamlit based UI.</br>
:red_circle: **Approach** : Building a streamlit UI and creating dropdowns for users to select the features and then predict the MRP of their laptop with the click of a single button.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Gaurav Kumar Singh
- GitHub Profile Link : https://github.com/Gaurav-576
- Email ID : gauravsingh96753@gmail.com
- Participant ID (if applicable):
- Approach for this Project : Streamlit based UI
- What is your participant role? (Mention the Open Source program) GSSoC'24

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
> /assign

Can you share your approach?
@aankiita 

P.S. We don't have that auto assign feature. Here every contributor share their approach and based on that the issues are getting assigned. Without a justification, this will not work here.

**Comment by Intelli2Byte** [comment]
# Project Proposal

**Full Name:** Neha Maurya
**GitHub Profile:** https://github.com/Intelli2Byte
**Email ID:** mauryaneha2006@gmail.com
**Participant Role:** GSSoC 2026 Contributor

## Project Title
**Real-Time Laptop Price Prediction using Web Scraping, Machine Learning & Streamlit**

## Aim
To build an interactive Streamlit web application that predicts laptop prices using real-world market data collected through web scraping. The platform will help users estimate fair laptop prices, compare configurations, and understand the key factors influencing pricing.

## Approach
### Phase 1 — Real-Time Data Collection

* Scrape laptop listings using BeautifulSoup and Selenium.
* Collect:
  * Brand
  * Processor
  * RAM
  * Storage (SSD/HDD)
  * GPU
  * Screen Size
  * Display Type
  * Operating System
  * Market Price
* Perform data cleaning, duplicate removal, and feature standardization.

### Phase 2 — Feature Engineering

* Extract CPU brand and generation.
* Extract GPU brand.
* Parse SSD and HDD capacities separately.
* Calculate PPI from screen size and resolution.
* Convert RAM and Weight into numerical values.
* Apply log transformation on price to reduce skewness.

### Phase 3 — Model Training

Models to be compared:
1. Linear Regression
2. Ridge Regression
3. Random Forest Regressor
4. Gradient Boosting Regressor
5. XGBoost Regressor

**Evaluation Metrics:** R², MAE, RMSE, and 5-Fold Cross Validation.

The best-performing model will be saved as a complete Scikit-Learn Pipeline for reliable inference.

### Phase 4 — Streamlit Application
**User Inputs:**

* Brand
* CPU
* RAM
* Storage
* GPU
* OS
* Screen Size
* Display Type

**Outputs:**

* Predicted Laptop Price (₹)
* Confidence Range (±10%)
* Feature Importance Visualization
* Model Comparison Table
* Similar Laptop Recommendations
* Market Analytics Dashboard

### Phase 5 — Documentation

Files:

* scraper.py
* preprocessing.py
* train.py
* app.py
* utils.py

Documentation will include:
* Project Overview
* Data Collection Process
* EDA Visualizations
* Model Comparison
* Screenshots
* Setup & Execution Guide

## Project Structure
```text
Real-Time-Laptop-Price-Prediction/
├── Dataset/
│   └── README.md
├── Images/
│   ├── price_distribution.png
│   ├── correlation_heatmap.png
│   ├── feature_importance.png
│   ├── market_trends.png
│   └── app_screenshot.png
├── Model/
│   ├── scraper.py
│   ├── preprocessing.py
│   ├── train.py
│   ├── app.py
│   ├── utils.py
│   ├── laptop_price_pipeline.pkl
│   ├── requirements.txt
│   └── README.md
```

## Key Highlights
* Uses real-time scraped market data instead of a static dataset.
* End-to-end machine learning pipeline with no preprocessing mismatch.
* Similar laptop recommendation system.
* Cross-validated model evaluation.
* Interactive market analytics dashboard.


**Thank you for considering my proposal. I look forward to contributing to this project through GSSoC 2026 and building a practical, real-world machine learning application. 🚀**

**Comment by abhisheks008** [comment]
Hi @Intelli2Byte Laptop Price Prediction is already an existing project but it does not have an web app for realtime user experience. This issue is about the creation of the web app based on the existing project, not the full project.

Existing project: https://github.com/abhisheks008/DL-Simplified/tree/main/Laptop%20Price%20Prediction%20Using%20DL

**Comment by Intelli2Byte** [comment]
Hello @abhisheks008,

Thank you for the clarification. I understand that the Laptop Price Prediction model already exists in the repository and the scope of this issue is to build a Streamlit web application around the existing project rather than recreating the complete ML pipeline.

### Planned Contribution

* Integrate the existing trained model into a Streamlit application.
* Create an interactive and user-friendly interface for laptop price prediction.
* Develop input forms based on the model's required features.
* Display predicted laptop prices in real time.
* Add basic visualizations and insights to improve user experience.
* Improve project documentation and provide setup instructions.

### Implementation Plan

1. Analyze the existing model, dataset, and preprocessing workflow.
2. Build the Streamlit frontend and connect it with the trained model.
3. Test predictions and ensure compatibility with the current project.
4. Enhance UI/UX and update documentation.

I would like to work on this issue under GSSoC 2026. Looking forward to your feedback.

Thank you!
